api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -249,6 +251,11 @@ type FelixConfigurationSpec struct {
 	HealthEnabled *bool   `json:"healthEnabled,omitempty"`
 	HealthHost    *string `json:"healthHost,omitempty"`
 	HealthPort    *int    `json:"healthPort,omitempty"`
+	// HealthTimeoutOverrides allows the internal watchdog timeouts of individual subcomponents to be
+	// overriden; example: "internal-dataplane-main-loop=30s,calculation-graph=2m".  This is useful for
+	// working around "false positive" liveness timeouts that can occur in particularly stressful workloads
+	// or if CPU is constrained.  For a list of active subcomponents, see Felix's logs.
+	HealthTimeoutOverrides string `json:"healthTimeoutOverrides,omitempty" validate:"omitempty,keyDurationList"`
```

Comment:
Shall we make the the field a type which contains timeout values for all active components? 
```
type HealthTimeoutOverrides struct {
    CalculationGraph *metav1.Duration
    ....
```

The benefit is that the API is much clearer to user, and it you get the default FelixConfiguration, it will show you all the default values for timeout.

Reply:
I should be able to make it a list or a map but I'm reluctant to pre-define all the health report names since they're currently allocated in a dispersed way.  I can't easily write a UT to ensure that none are missed either since the ones that are active depend on build and run-time configuration.  

I envisioned this as a safety valve that we'd tell a user to set in an emergency rather than something that users would necessarily discover on their own.

Reply:
(The main reason I didn't do it was naming convention mismatch with k8s; seemed a bit odd to use `lowerCamel` for health report names. Could live with UpperCamel, as Casey suggested.)

Reply:
> pre-define all the health report names since they're currently allocated in a dispersed way

I think I'm begrudgingly OK with the argument that this is an escape valve that is meant more for developers to help customers than it is for users to set themselves. The problem is that we mix those sorts of fields with proper user-facing fields in the same resource so it's not super obvious which are which.

Reply:
I made this a slice of override objects and improved the documentation